rust/worker/src/server.rs

```diff
@@ -248,6 +250,23 @@ impl chroma_proto::vector_reader_server::VectorReader for WorkerServer {
 
 #[tonic::async_trait]
 impl chroma_proto::metadata_reader_server::MetadataReader for WorkerServer {
+    async fn count_records(
+        &self,
+        request: Request<CountRecordsRequest>,
+    ) -> Result<Response<CountRecordsResponse>, Status> {
+        let request = request.into_inner();
+        let segment_uuid = match Uuid::parse_str(&request.segment_id) {
+            Ok(uuid) => uuid,
+            Err(_) => {
+                return Err(Status::invalid_argument("Invalid Segment UUID"));
+            }
+        };
+        println!("Querying count for segment {}", segment_uuid);
+        // TODO: Add logic here to count.
```

Comment:
assuming this will happen in follow up PR?

Reply:
Yep